tests/subsys/fs/fs_api/src/test_fs_dir_file.c

```diff
@@ -94,17 +152,53 @@ void test_mount(void)
 	zassert_not_equal(ret, 0, "Mount to a mounted dir");
 
 	fs_unregister(TEST_FS_2, &temp_fs);
-
 	fs_register(TEST_FS_2, &null_fs);
+
 	TC_PRINT("Mount a file system has no interface implemented\n");
 	ret = fs_mount(&test_fs_mnt_no_op);
 	zassert_not_equal(ret, 0, "Mount to a fs without op interface");
-	fs_unregister(TEST_FS_2, &null_fs);
+	ret = fs_unmount(&test_fs_mnt_no_op);
+
+	/* mount an unmountable file system */
+	null_fs.mount = temp_fs.mount;
+	ret = fs_mount(&test_fs_mnt_no_op);
+	zassert_equal(ret, 0, "Mount to a fs without unmount interface");
```

Comment:
"interface"->"functionality"; interface is the entire FS API provided by fs driver.